tests/h/views/admin_users_test.py

```diff
@@ -265,6 +266,19 @@ def test_delete_user_deletes_annotations(api_storage, db_session, factories, pyr
 
 
 @delete_user_fixtures
+def test_delete_user_publishes_event(api_storage, db_session, factories, matchers, pyramid_request):
+    pyramid_request.db = db_session
+    user = factories.User()
+    ann = factories.Annotation(userid=user.userid)
+
+    delete_user(pyramid_request, user)
+
+    event = AnnotationEvent(pyramid_request, ann.id, 'delete')
+    pyramid_request.notify_after_commit.assert_called_once_with(
+        matchers.same_dicts(event))
```

Comment:
I'm not sure if comparing `__dict__` is a good idea, it works for this occasion but I can see some weird issues popping up in the future. According to the Python doc `__dict__` returns "a dictionary or other mapping object used to store an object’s (writable) attributes." - Technically this doesn't ensure that it is the same object, just that it has the same writable attributes.
Furthermore, I don't think that anybody who didn't first see the implementation of the matcher will know what `same_dicts` does.

I think it might be better and more readable, although also more verbose, to do this:

```
@delete_user_fixtures
def test_delete_user_publishes_event(api_storage, db_session, factories, matchers, pyramid_request):
    pyramid_request.db = db_session
    user = factories.User()
    ann = factories.Annotation(userid=user.userid)

    delete_user(pyramid_request, user)

    event = pyramid_request.notify_after_commit.call_args[0][0]
    assert event.annotation_id == ann.id
    assert event.action == 'delete'
```
At least this is using `mock`'s `call_args`, and the check that the event is the same is more explicit.